src/partialBind.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/prefer-readonly-parameter-types, @typescript-eslint/no-unsafe-return --
+ * Function inference doesn't work when `unknown` is used as the parameters
+ * generic type, it **has** to be `any`.
+ */
+import type { IterableContainer, RemedaTypeError } from "./internal/types";
+
+type PartialBindError<Message extends string> = [
+  RemedaTypeError<"partialBind", Message>,
+];
+
+type RemovePrefix<
+  T extends IterableContainer,
+  Prefix extends IterableContainer,
+> = Prefix["length"] extends 0
+  ? T
+  : T["length"] extends 0
+    ? PartialBindError<"Too many args provided to function">
+    : T extends [infer THead, ...infer TRest]
```

Comment:
Even if effectively T can never be readonly, we should make the type itself as generic as possible, especially considering its name considers it as a generic util. mutable tuples always extend the readonly version with the same shape.
```suggestion
    : T extends readonly [infer THead, ...infer TRest]
```